src/elf.c

```diff
@@ -14,7 +14,14 @@
 
 static unsigned long uncache_to_cache(const struct image *image, unsigned long address)
 {
-	return ((address & ~image->adsp->alias_mask) | image->adsp->alias_cached);
+	unsigned long base = image->adsp->mem_zones[SOF_FW_BLK_TYPE_ROM].base;
+	size_t size = image->adsp->mem_zones[SOF_FW_BLK_TYPE_ROM].size;
```

Comment:
Hmm, I think a bit of background would be needed either in commit or code comments. To start with, I don't fully understand "we don't send ROM data to DSP"  in this context. I mean surely the DSP is consuming ROM data as well, it's just we don't put the put ROM data into the FW main image (that we are producing here). So maybe just rephrase a bit.

Then I'm wondering could this ROM check be at a higher level. This is a bit surprising that the uncache_to_cache() has this logic inside. But I guess you looked at different options and identifying the ROM bits is not doable in the caller...?


But then I wonder could this check be on a higher level